sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/InputStreamContent.java

```diff
@@ -8,23 +8,32 @@
 import com.azure.core.util.serializer.ObjectSerializer;
 import com.azure.core.util.serializer.TypeReference;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
 
+import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.SequenceInputStream;
 import java.io.UncheckedIOException;
 import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
 import java.util.Objects;
+import java.util.Vector;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
 
 /**
  * A {@link BinaryDataContent} implementation which is backed by an {@link InputStream}.
  */
 public final class InputStreamContent extends BinaryDataContent {
     private static final ClientLogger LOGGER = new ClientLogger(InputStreamContent.class);
-    private final InputStream content;
+    private static final int BUFFER_CHUNK_SIZE = 8 * 1024 * 1024;
```

Comment:
Arbitrary size?

Reply:
I definitely agree that 8KB buffers would have much too large of a % overhead with just wrangling the array/InputStream, but maybe split the difference a bit with either 1MB or 4MB chunks? With both overhead would be negligible and we're reducing the case you mention with a 100MB stream where the last buffer will be half full, wasting 4MB of heap (which would likely be larger than the overhead of many, many 8KB buffers with a rough math of 256 bytes overhead per chunk that'd be ~3MB in overhead). This is also something we could attempt to make variable if the InputStream size is known, for example if it's <8MB we could read into a single buffer, if its >1GB maybe we push to 16MB chunks.

Pushing this discussion further, we may want to look into making these default buffer sizes consistent across Core. I know in some other areas we chunk in 4KB and 8KB increments and we'd likely see benefits increasing them to something larger like 64KB or 128KB.

Reply:
Good points about known length I have made algo bit smarter.